src/core/Pagination/PageInput/PageInput.tsx

```diff
@@ -0,0 +1,259 @@
+      }
+      setValue('');
+    };
+
+    const onKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (
+        !!this.state.value &&
+        this.state.status !== 'error' &&
+        event?.key === 'Enter'
+      ) {
+        onPageChange();
+      }
+    };
+
+    const onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event?.key === 'Escape') {
+        event.preventDefault();
+      }
+    };
+
+    const buttonClick = () => {
+      /* Get the current input value */
+      const tempValue = this.state.value;
+      /* Clear the input field so screenreader wont read the "old" number again */
+      setValue('');
+
+      /**
+       * When the action comes from button it still has the focus. Button will be hidden when input (and state)
+       * are cleared so the focus must be moved back to the input. Otherwise screen readers (NVDA) will have focus issues.
+       * Timeout is needed so the input field has enough time to clear.
+       * Without this the screenreader is too fast and reads the old number.
+       */
+      setTimeout(() => {
+        if (this.inputRef.current) {
+          this.inputRef.current.focus();
+        }
+      }, 100);
+
+      /* Call prop with the new value after a timeout. Now the "page change text" is the last one to be read.
+      Without the timeout NVDA skips it.
+      */
+      setTimeout(() => {
+        if (!!propOnPageChange) {
+          propOnPageChange(tempValue);
+        }
+      }, 200);
+    };
+
+    const pageInputButtonDerivedProps = {
+      className: pageInputClassNames.button,
+      ...(!!this.state.value && this.state.status !== 'error'
+        ? { onClick: buttonClick }
+        : { tabIndex: -1, hidden: true }),
+    };
+
+    return (
+      <HtmlDiv
+        className={classnames(className, baseClassName, {
+          [pageInputClassNames.error]: this.state.status === 'error',
+          [pageInputClassNames.notEmpty]:
+            !!this.state.value && this.state.status !== 'error',
+        })}
+      >
+        <HtmlSpan className={pageInputClassNames.styleWrapper}>
+          <Label htmlFor={id} labelMode="hidden">
+            {labelText}
+          </Label>
+          <HtmlDiv className={pageInputClassNames.functionalityContainer}>
+            <HtmlDiv className={pageInputClassNames.inputElementContainer}>
+              <HtmlInput
+                {...passProps}
+                {...getConditionalAriaProp('aria-describedby', [
+                  !!statusText ? statusTextId : undefined,
+                  ariaDescribedBy,
+                ])}
+                size={6}
+                aria-invalid={this.state.status === 'error'}
+                id={id}
+                forwardedRef={this.inputRef}
+                className={pageInputClassNames.inputElement}
+                value={this.state.value}
+                placeholder={visualPlaceholder}
+                onChange={(event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
Maybe onChange function block could be in its own function like onKeyPress and onKeyDown